main.tf

```
@@ -1,9 +1,126 @@
+# Documentation at https://codeberg.org/hjacobs/kube-downscaler
+
 terraform {
-  required_version = ">= 1.3.0"
+  required_version = ">= 1.5.0"
+
   required_providers {
-    aws = {
-      source  = "hashicorp/aws"
-      version = ">= 4.0"
+    kubernetes = {
+      source  = "hashicorp/kubernetes"
+      version = "~> 2.0"
+    }
+  }
+}
+
+
+locals {
+  downscaler_name = "downscaler"
+  downscaler_labels = {
+    app = local.downscaler_name
+  }
+}
+
+resource "kubernetes_namespace_v1" "kube_downscaler" {
```

minor:

```suggestion
resource "kubernetes_namespace_v1" "this" {
```
because of https://www.terraform-best-practices.com/naming#resource-and-data-source-arguments